From 0af7230d9bc67ce4ab81552157272cf8160d36dc Mon Sep 17 00:00:00 2001
From: Karl Wette <karl.wette@ligo.org>
Date: Mon, 27 Oct 2025 16:33:19 +1100
Subject: [PATCH 2/3] lal/swig/SWIGPython.i: use import_array1(0) to import
 NumPy if available

- import_array() returns NULL on error which conflicts with the SWIG
  function it is called from
---
 lal/swig/SWIGPython.i | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lal/swig/SWIGPython.i b/lal/swig/SWIGPython.i
index 8f332822d1..63209957e2 100644
--- a/lal/swig/SWIGPython.i
+++ b/lal/swig/SWIGPython.i
@@ -62,7 +62,11 @@
 #endif
 %}
 %init %{
+#if defined(import_array1)
+import_array1(0);
+#else
 import_array();
+#endif
 %}
 
 // Evaluates true if a PyObject is not empty, false otherwise.
-- 
2.39.5

